File: app/src/main/main/java/com/timeback/ui/feature/datamanagement/DataManagementViewModel.java
package com.timeback.ui.feature.datamanagement;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.timeback.ui.domain.gateway.FeatureGateway;
import com.timeback.ui.domain.model.*;

import javax.inject.Inject;
import dagger.hilt.android.lifecycle.HiltViewModel;

/**
 * UI-08 데이터 관리 ViewModel.
 * 보존 기간 변경, 전체 삭제 요청/확인, 삭제 상태 확인을 처리한다.
 */
@HiltViewModel
public class DataManagementViewModel extends ViewModel {

    private final FeatureGateway gateway;
    private final MutableLiveData<ScreenState> _state = new MutableLiveData<>(new ScreenState.Initial());
    public final LiveData<ScreenState> state = _state;

    private final MutableLiveData<ActionResult> _actionResult = new MutableLiveData<>();
    public final LiveData<ActionResult> actionResult = _actionResult;

    @Inject
    public DataManagementViewModel(FeatureGateway gateway) {
        this.gateway = gateway;
        loadState();
    }

    public void loadState() {
        _state.setValue(new ScreenState.Loading());
        gateway.readDataManagementState().thenAccept(_state::postValue);
    }

    public void changeRetention(RetentionSelection selection) {
        gateway.changeRetentionSelection(selection).thenAccept(_actionResult::postValue);
    }

    public void requestDeletion() {
        gateway.requestFullDeletion().thenAccept(_actionResult::postValue);
    }

    public void confirmDeletion() {
        gateway.confirmFullDeletion().thenAccept(result -> {
            _actionResult.postValue(result);
            if (result instanceof ActionResult.Success) {
                loadState();
            }
        });
    }

    public void refreshDeletionStatus(Identifier jobId) {
        gateway.refreshDeletionStatus(jobId).thenAccept(_state::postValue);
    }
}

File: src/main/java/com/timeback/ui/feature/datamanagement/DataManagementViewModel.kt
package com.timeback.ui.feature.datamanagement

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.timeback.ui.domain.gateway.FeatureGateway
import com.timeback.ui.domain.model.*
import dagger.hilt.android.lifecycle.HiltViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch
import javax.inject.Inject

/**
 * UI-08 데이터 관리 ViewModel
 * CT-04 §11: ReadDataManagementState, ChangeRetentionSelection, RequestFullDeletion, ConfirmFullDeletion, RefreshDeletionStatus
 */
@HiltViewModel
class DataManagementViewModel @Inject constructor(
    private val gateway: FeatureGateway
) : ViewModel() {

    private val _state = MutableStateFlow<ScreenState>(ScreenState.Initial)
    val state: StateFlow<ScreenState> = _state.asStateFlow()

    private val _actionResult = MutableStateFlow<ActionResult?>(null)
    val actionResult: StateFlow<ActionResult?> = _actionResult.asStateFlow()

    private val _confirmationPending = MutableStateFlow(false)
    val confirmationPending: StateFlow<Boolean> = _confirmationPending.asStateFlow()

    init {
        loadDataManagement()
    }

    fun loadDataManagement() {
        viewModelScope.launch {
            _state.value = ScreenState.Loading
            _state.value = gateway.readDataManagementState()
        }
    }

    fun changeRetention(selection: RetentionSelection) {
        viewModelScope.launch {
            _state.value = ScreenState.Loading
            val result = gateway.changeRetentionSelection(selection)
            _actionResult.value = result
            if (result is ActionResult.Success) {
                _state.value = gateway.readDataManagementState()
            } else {
                _state.value = gateway.readDataManagementState()
            }
        }
    }

    fun requestFullDeletion() {
        viewModelScope.launch {
            val result = gateway.requestFullDeletion()
            if (result is ActionResult.Success) {
                _confirmationPending.value = true
            }
            _actionResult.value = result
        }
    }

    fun confirmFullDeletion() {
        viewModelScope.launch {
            _confirmationPending.value = false
            _state.value = ScreenState.Loading
            val result = gateway.confirmFullDeletion()
            _actionResult.value = result
            if (result is ActionResult.Success) {
                _state.value = gateway.readDataManagementState()
            }
        }
    }

    fun cancelDeletion() {
        _confirmationPending.value = false
    }

    fun refreshDeletionStatus(jobId: Identifier) {
        viewModelScope.launch {
            _state.value = gateway.refreshDeletionStatus(jobId)
        }
    }

    fun clearActionResult() {
        _actionResult.value = null
    }
}
